### ISSUE TITLE
Migrate our js files away from using jquery

### ISSUE THREAD

**Issue by parlough** [flutter/website#8889]
### Describe the problem

We'd eventually like to stop using jquery, especially since Bootstrap 5 ([which we hope to migrate to](https://github.com/flutter/website/issues/8030)) no longer requires it. 

### Expected fix

Migrate all code in https://github.com/flutter/website/tree/main/src/assets/js away from using jquery.

**Comment by parlough** [comment]
If anyone wants to help with this, please keep your pull requests simple. Migrate a few related functions, or if a file is simple, just one file. Thanks so much!

**Comment by ulisseshen** [comment]
It may not be an easy job, but I appreciate it very much and I can try to help with that. However, I have no previous experience collaborating with open source projects.

**Comment by parlough** [comment]
We always appreciate help! I'm happy to help with any issues you have as well.

The repository's [README](https://github.com/flutter/website/tree/main#flutter-website) has some information on contributing and setting up the site. 

Usually contributing looks like some mix of [forking the repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo), making and testing your changes in a [local branch](https://gist.github.com/Chaser324/ce0505fbed06b947d962#doing-your-work), then pushing them to GitHub and opening a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request). From there one of us will review it and provide feedback in a back and forth process.

As for this particular issue, perhaps a good place to start would be a single function (or even a part of one) in the [`archive.js`](https://github.com/flutter/website/blob/main/src/assets/js/archive.js) folder. It is used on the [Flutter SDK archive page](https://docs.flutter.dev/release/archive) for populating the archive tables. There are a lot of good documents online about migrating away from jquery to normal JavaScript.

If you'd like to work on that, go for it, and let us know if you have any questions about anything or something doesn't work.

**Comment by ulisseshen** [comment]
I already hosted this project locally, I have no problem with it. A good starting point could be this [archive.js](https://github.com/flutter/website/blob/main/src/assets/js/archive.js) file you mentioned.

My questions now are:
How can I relate my changes to this topic?
Will Jquery's ajax calls be replaced by the fetch api?

PS: Good initiative. As awesome as it is, the Jquery era has long since passed.

**Comment by parlough** [comment]
@ulisseshen 

> How can I relate my changes to this topic?

If I'm working on part of an issue, generally what I do is create a pull request against this repository with my changes, then in the pull request description add text like `Contributes to https://github.com/flutter/website/issues/8889`. If my pull request will completely fix the issue, I instead write `Fixes https://github.com/flutter/website/issues/8889` as it will then close the issue automatically when the pull request is merged.

> Will Jquery's ajax calls be replaced by the fetch api?

Yeah, I think that sounds good. We don't support Internet Explorer and all other major browsers have [supported it for a while](https://caniuse.com/fetch).

> PS: Good initiative. As awesome as it is, the Jquery era has long since passed.

Yeah for sure :) - Hopefully this will make our little JS code easier to maintain while also being quicker to download.

**Comment by ulisseshen** [comment]
Soon I'll continue working on this. I feel grateful for give some help.

**Comment by parlough** [comment]
@ulisseshen I'll assign you then but take your time :) 

Thanks again for your help!

**Comment by ulisseshen** [comment]
@parlough while I was making some changes to the [archive.js](https://github.com/flutter/website/blob/main/src/assets/js/archive.js) file, there is a bit of code that is runs in initial page load. Above a code that I'm talking about:
https://github.com/flutter/website/blob/15bc68f068b7eeea4fae16d9f9093b36ff794f1a/src/assets/js/archive.js#L246-L258

I belive that the code between line 252 and 257 is not used anymore. I did some test to validate this, I it look that is not used. I will not migrate thie code betwwen those lines and not funcitons `updateDownloadLink` and `updateDownloadLinkFailed`, that are used only in this call.

**Comment by parlough** [comment]
Thanks for looking into it!

I believe the code you linked to is used on the "Windows install", "macOS install", and similar pages to set up the download buttons for the latest release on those pages. An example is here: https://docs.flutter.dev/get-started/install/windows#get-the-flutter-sdk

The source for that button is https://github.com/flutter/website/blob/main/src/get-started/install/_get-sdk-win.md?plain=1#L10

Perhaps we should add some comments to the code to make that more clear, because it is pretty easy to miss.

**Comment by ulisseshen** [comment]
You are rigth. I ended up discovering this while invetegateding a little more. I will continue then migrating this code and i will make some comments to improve understanding by others in the future

**Comment by ulisseshen** [comment]
In many cases JQuey return a list of elements in your selector and when apply any input in those elements selected it do it in all elements at once. 

```
// Will change every element that have a class `my-class`
$('.my-class').text('my newest text')
```

To migrate to pure js have two ways. 

Considered that will have only a single element and to use `document.querySelector` or that will have one or more elements and to use `document.querySelectorAll`.

For this fase of migration I have considerated that always will be one or more elements. So, I used `document.querySelectorAll`
